lib/Item.php

```diff
@@ -190,23 +198,36 @@ public function processUnchecked(Status $status, $isBackground) {
 	 * @param boolean $isBackground
 	 */
 	public function processClean(Status $status, $isBackground) {
-		if (!$isBackground) {
+		if (!$isBackground || $this->id === null) {
 			return;
 		}
 		try {
-			$stmt = \OCP\DB::prepare('DELETE FROM `*PREFIX*files_antivirus` WHERE `fileid` = ?');
-			$result = $stmt->execute([$this->id]);
-			if (\OCP\DB::isError($result)) {
-				//TODO: Use logger
-				$this->logError(__METHOD__. ', DB error: ' . \OCP\DB::getErrorMessage());
-			}
-			$stmt = \OCP\DB::prepare(
-				'INSERT INTO `*PREFIX*files_antivirus` (`fileid`, `check_time`) VALUES (?, ?)'
+			$dbConnection = \OC::$server->getDatabaseConnection();
+			$delete = $dbConnection->getQueryBuilder();
```

Comment:
Why use delete + insert instead of just update the row?. If the expectation is that the file will likely exists (the cases where the file will be missing are quite a few), you might want to try an update and if nothing updates then insert. This should perform better since most of the time you'll hit the DB once instead of twice.

Reply:
As for now I just rewrote the queries to use query builder. But if you don't like it I can revert this change to call `OC_DB` ;)

Reply:
I'm fine if this change is done later as long as there is a ticket for that.

Reply:
np, we have upsert in 10.0.9. This part has been rewritten to use it.